ddd-koans/Value Object/Summary & Data Class/src/Main.kt

```diff
@@ -0,0 +1,16 @@
+sealed class TimeSignature
+
+data class ValidTimeSignature(private val numerator: Int, private val denominator: Int) : TimeSignature() {
+    init {
+        require(numerator in 1..32) {
+            "Numerator must be integer between 1 and 32"
+        }
+        require(denominator in 1..32 && denominator.isPowerOfTwo()) {
```

Comment:
shouldn't we ask rather to implement a full solution here using data class including static-factory, returning Valid/Invalid type and `of("3/4")` function to use domain-specific notation?

Reply:
I misunderstood then what we were going to implement as I thought this summary and data class task would be just edu (without anything to code) and usage of data class would be in the next task. Do you mean task where you have to implement everything except that that is covered by using data class?

Reply:
exactly but we can decide what should be the outcome of this task 😉